sw/device/lib/dif/dif_otbn.c

```diff
@@ -234,8 +231,8 @@ dif_otbn_result_t dif_otbn_imem_write(const dif_otbn_t *otbn,
                                       uint32_t offset_bytes, const void *src,
                                       size_t len_bytes) {
   // Only 32b-aligned 32b word accesses are allowed.
-  if (otbn == NULL || src == NULL || len_bytes % 4 != 0 ||
-      offset_bytes % 4 != 0 ||
+  if (otbn == NULL || src == NULL || len_bytes % sizeof(uint32_t) != 0 ||
+      offset_bytes % sizeof(uint32_t) != 0 ||
       offset_bytes + len_bytes > OTBN_IMEM_SIZE_BYTES) {
     return kDifOtbnBadArg;
```

Comment:
Doesn't this also need a check for overflow of `offset_bytes + len_bytes`? I wonder whether it would make sense to define a helper function for this? Something like this, maybe?
```c
static int check_offset_len(uint32_t offset_bytes, size_t len_bytes, size_t mem_size) {
  return (len_bytes % sizeof(uint32_t) == 0 &&
          offset_bytes % sizeof(uint32_t) == 0 &&
          offset_bytes + len_bytes >= len_bytes &&
          offset_bytes + len_bytes <= mem_size);
}
```

Reply:
Like the third line in the if statement (`offset_bytes + len_bytes > OTBN_IMEM_SIZE_BYTES`) :smile: But yeah, I can put it into a helper function.

Reply:
> Ah, you mean an integer overflow in the addition.

Yep!

Reply:
Tip: this should return `bool`, it's allowed :) 

And that overflow check is only valid because both inputs are unsigned. That should be documented. Detecting signed overflow is much more difficult.

Reply:
Good points, updated.